Change to eclipse project

sgraband opened the pull request:
Signed-off-by: Simon Graband <sgraband@eclipsesource.com>

Co-authored-by: Tobias Ortmayr <tortmayr@eclipsesource.com>

sgraband committed:
Change to eclipse project

Signed-off-by: Simon Graband <sgraband@eclipsesource.com>

Co-authored-by: Tobias Ortmayr <tortmayr@eclipsesource.com>

eneufeld commented on pom.xml:
```diff
@@ -0,0 +1,357 @@
+	<groupId>org.eclipse.emfcloud.validation</groupId>
+	<artifactId>org.eclipse.emfcloud.validation.parent</artifactId>
+	<description>Model Validation Parent Pom</description>
+	<version>1.0.0-SNAPSHOT</version>
```
we should not use 1.0.0, it should rather be something like 0.2.0 or so

eneufeld commented on pom.xml:
```diff
@@ -0,0 +1,357 @@
+									<arch>x86_64</arch>
+								</environment>
+							</environments>
+							<target>
```
you can specify the target file directly using tycho 2.3: https://wiki.eclipse.org/Tycho/Release_Notes/2.3#Target_files_can_be_specified_directly_now

before 2.3 it is also not needed to have a classifier, the target file just needs to be named as the artifactid see https://github.com/eclipse-emfcloud/coffee-editor/tree/master/backend/releng/org.eclipse.emfcloud.coffee.target

eneufeld commented on pom.xml:
```diff
@@ -0,0 +1,357 @@
+			<activation>
+				<activeByDefault>false</activeByDefault>
+			</activation>
+			<modules>
```
why can't we move the modules outside the profiles? they are the same for all builds, aren't they?

eneufeld commented on releng/org.eclipse.emfcloud.validation.codestyle/checkstyle/model-validation-checkstyle.xml:
```diff
@@ -0,0 +1,276 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
can't we reuse the checkstyle from Tobias: https://github.com/eclipse-emfcloud/emfcloud/tree/master/codestyle

eneufeld commented on targetplatform/r2021-03.tpd:
```diff
@@ -0,0 +1,23 @@
+	org.eclipse.emf.sdk.feature.group
+}
+
+location "http://ghillairet.github.io/p2" {
```
@planger didn't we publish a version of emfjson since we moved it to eclipse?

planger replied:
No we didn't publish an emfjson release on the Eclipse infra yet :/
I think we'll have to stick with this for now here.

eneufeld reviewed (changes requested):
Mainly the version should be changed to something other than 1.0.0 as we won't have a release of the framework just yet. To be in eclipse incubation we need a version below 1.0.0 .
Looks great otherwise!

sgraband committed:
Use central checkstyle config

- Use central checkstyle config from
https://github.com/eclipse-emfcloud/emfcloud/tree/master/codestyle
- Change Version to 0.2.0-SNAPSHOT
- Directly reference targetfile
- Upgrade to tycho 2.3.0

Signed-off-by: Simon Graband <sgraband@eclipsesource.com>

sgraband committed:
Update to newest modelserver version

Signed-off-by: Simon Graband <sgraband@eclipsesource.com>

eneufeld commented on pom.xml:
```diff
@@ -0,0 +1,355 @@
+			</repositories>
+			<modules>
+				<module>org.eclipse.emfcloud.validation.framework</module>
+				<module>targetplatform</module>
```
why do we need the targetplatform in the m2 build?

sgraband replied:
I removed it.

eneufeld reviewed (approved):
Looks good. As we need to wait before we can start the cq, I suggest to wait for Philip to respond

sgraband committed:
Remove targetplatform from m2 build

Signed-off-by: Simon Graband <sgraband@eclipsesource.com>
